### PR TITLE
MAINT: Fix a few typos - and sometimes improve wording

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Additional fixes from #27336 not in #27344.

### CODE REVIEW

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
The test failure looks unrelated.

**Comment by charris** [comment]
Thanks @DimitriPapadopoulos .

File: meson_cpu/main_config.h.in [hunk 1]
```diff
@@ -11,7 +11,7 @@
  */
 #ifndef @P@_CPU_DISPATCHER_CONF_H_
 #define @P@_CPU_DISPATCHER_CONF_H_
-/// This definition is required to provides comptablity with NumPy distutils
+/// This definition is required to provide comptibility with NumPy distutils
```
Comment (charris) [inline meson_cpu/main_config.h.in:14]: Hmm, I think this is still wrong. Missed it.

Comment (DimitriPapadopoulos) [reply meson_cpu/main_config.h.in:14]: Ooops... I will run codespell again at some point, it should catch this typo.

**Comment by musvaage** [comment]
numpy/_core/src/multiarray/buffer.c

I presume `requirements` is contextually adequate or a better replacement string  than the ticket's suggestion.

```diff
+  * Implementing PEP 3118 is somewhat convoluted because of the priorities:
```

```diff
-  * Implementing PEP 3118 is somewhat convoluted because of the desirata:
+  * Implementing PEP 3118 is somewhat convoluted because of the requirements:
```

numpy/lib/\_\_init\_\_.py

I presume the implemented replacement line is a better fit than the ticket's more verbose suggestion.

```diff
+     # Warn for deprecated/removed aliases, now private and nonexistent attributes
```

```diff
-     # Warn for reprecated attributes
+     # Warn for deprecated/removed aliases
```

<hr>

_this is only for my records_

doc/release/upcoming_changes/27156.change.rst
numpy/f2py/tests/test_array_from_pyobj.py

In this pull a typo was caught in each of the above files which the work on the linked to ticket had overlooked.

**Comment by musvaage** [comment]
> run codespell again

```
$ pushd numpy/numpy/_core/src/common
~/numpy/numpy/_core/src/common ~
$ grep -nr de-interlave simd && popd
simd/sse/memory.h:686: *  de-interlave load / interleave contiguous store
simd/avx2/memory.h:708: *  de-interlave load / interleave contiguous store
simd/vec/memory.h:626: *  de-interlave load / interleave contiguous store
simd/neon/memory.h:587: *  de-interlave load / interleave contiguous store
simd/avx512/memory.h:654: *  de-interlave load / interleave contiguous store
~
$ grep -n "discret " numpy/numpy/lib/_function_base_impl.py
65:#   A function used for discret methods to force the index to a specific value.
$ grep -n _discret numpy/numpy/lib/_function_base_impl.py
4636:def _discret_interpolation_to_boundaries(index, gamma_condition_fun):
4654:    return _discret_interpolation_to_boundaries((n * quantiles) - 1 - 0.5,
4660:    return _discret_interpolation_to_boundaries((n * quantiles) - 1,
$ grep -n fastests numpy/.github/workflows/linux.yml
277:        # Install using the fastests way to build (no BLAS, no SIMD)
$ grep -n whiher numpy/meson_cpu/main_config.h.in
49: * using preprocessor callback without testing whiher the
$ 
```

**Comment by DimitriPapadopoulos** [comment]
I am not sure how to fix this, doesn't make any sense to me:
```
$ grep -n wither numpy/numpy/_core/tests/test_half.py 
52:        # avoid testing NaNs due to differ bits wither Q/SNaNs
$ 
```
Can you help?

**Pull request merged**